src/CoreWCF.Primitives/src/CoreWCF/Security/SecurityUtils.cs

```diff
@@ -319,6 +374,19 @@ internal static void EnsureCertificateCanDoKeyExchange(X509Certificate2 certific
             }
         }
 
+        public static WrappedKeySecurityToken CreateTokenFromEncryptedKeyClause(EncryptedKeyIdentifierClause keyClause, SecurityToken unwrappingToken)
+        {
+            SecurityKeyIdentifier wrappingTokenReference = keyClause.EncryptingKeyIdentifier;
+            byte[] wrappedKey = keyClause.GetEncryptedKey();
+            SecurityKey unwrappingSecurityKey = unwrappingToken.SecurityKeys[0];
+            string wrappingAlgorithm = keyClause.EncryptionMethod;
+            byte[] unwrappedKey = unwrappingSecurityKey.DecryptKey(wrappingAlgorithm, wrappedKey);
+            //TODO, check value for XmlDictionaryString Symmetric or else 
```

Comment:
WeappedKeySecurityToken is missing an internal constructor. You can find it [here](https://referencesource.microsoft.com/#System.IdentityModel/System/ServiceModel/Security/Tokens/WrappedKeySecurityToken.cs,71).

Reply:
coming with WIndowAuth...